app/api/register.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+class Register < Grape::API
+  format :json
+
+  desc 'Register Endpoint'
+  namespace :register do
+    desc 'Register with email and password'
+    params do
+      requires :email, type: String, desc: 'email'
+      requires :password, type: String, desc: 'password'
+    end
+    post do
+      user = User.new(email: params[:email], password: params[:password])
+      unless user.present? && user.valid_password?(params[:password]) && user.save
```

Comment:
Even though it might be less code this way, we might break the save and the validations sides. Having a block that first validates that the user is present and the password is valid. Then call the save method, `if user.save` returns 201, otherwise, we raise the error and 400.